### PR TITLE
Upgrade MY-WM parser

### PR DESCRIPTION
**Pull request by mathilde-daugy**
## Issue

image
We're getting too many error logs with the ADR and upgrading to fit the new parser scheme should help this 
## Description

Upgrade parser functions to new parser scheme. This includes:
- consumption
- exchanges
- production

### Preview

NA

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

**Review by unitrium — APPROVED** [review]
Looks good!

File: parsers/MY_WM.py [hunk 1]
```diff
@@ -29,65 +36,77 @@
...
+    zone_key2: ZoneKey,
+    session: Session = Session(),
     target_datetime: Optional[datetime] = None,
     logger: Logger = getLogger(__name__),
 ) -> list:
     """Request the power exchange (in MW) between two zones."""
     date_string = arrow.get(target_datetime).to(TIMEZONE).format("DD/MM/YYYY")
-    session = session or Session()
+
     sorted_zone_keys = "->".join(sorted((zone_key1, zone_key2)))
```
Comment (VIKTORVAV99) [inline parsers/MY_WM.py:89]:
```suggestion
    sorted_zone_keys = ZoneKey("->".join(sorted([zone_key1, zone_key2])))
```

File: parsers/MY_WM.py [hunk 2]
```diff
@@ -29,65 +36,77 @@
...
+    )
+    all_consumption_data = TotalConsumptionList(logger)
+
+    for item in consumption_data:
+        all_consumption_data.append(
+            zoneKey=zone_key,
+            datetime=arrow.get(item["DT"]).to(TIMEZONE).datetime,
+            consumption=item["MW"],
+            source=DOMAIN,
         )
```
Comment (VIKTORVAV99) [inline parsers/MY_WM.py:74]: Are we sure this is measured data? The future data points seems to indicate this is forcasted data instead.

File: parsers/MY_WM.py [hunk 3]
```diff
@@ -108,59 +127,49 @@ def fetch_exchange(
...
+            zoneKey=zone_key,
+            production=production_mix,
+            source=DOMAIN,
+            datetime=item_datetime,
         )
-    ]
+    return all_production_data.to_list()
 
 
 @config.refetch_frequency(timedelta(minutes=10))
```
Comment (VIKTORVAV99) [inline parsers/MY_WM.py:175]: Do you want to upgrade this function in the parser as well?
Since we are changing all the others I think it makes sense.

**Review by VIKTORVAV99 — COMMENTED** [review]
Just some small stuff.

**Comment by VIKTORVAV99** [comment]
Oh you guys are quick...

File: parsers/MY_WM.py (hunk 3 above)
Comment (mathilde-daugy) [reply parsers/MY_WM.py:175]: I didn't update this one because we don't make any use of production forecasts. But for consistency I will

**Pull request merged**